typing/env.ml

```diff
@@ -584,54 +684,75 @@ let find_same_module id tbl =
   | x -> x
   | exception Not_found
     when Ident.persistent id && not (Current_unit_name.is_name_of id) ->
-      Persistent
+      Mod_persistent
 
-(* signature of persistent compilation units *)
-type persistent_module = {
-  pm_signature: signature Lazy.t;
-  pm_components: module_components;
-}
+let find_name_module ~mark name tbl =
+  match IdTbl.find_name wrap_module ~mark name tbl with
+  | x -> x
+  | exception Not_found when not (Current_unit_name.is name) ->
```

Comment:
I don't think so. That was a weird discrepancy between `lookup_module` and `lookup_module_descr` in the old code. It meant that in theory `M.t` could sometimes be resolved when `M` could not, although in practice I don't think this would have happened except in very strange corner cases involving parallel builds. The new code is more obviously consistent.